src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -458,6 +461,18 @@ private void PreInitialize()
                 _metricsLogger.LogEvent(MetricEventNames.ApplicationInsightsDisabled);
             }
 
+            if (!string.IsNullOrEmpty(_settingsManager.TelemetryMode) && _settingsManager.TelemetryMode.Equals(OpenTelemetryConstants.OpenTelemetry, StringComparison.OrdinalIgnoreCase))
```

Comment:
You have an enum for `TelemetryMode` already, can settings manager use that as the type?